commit f7c4ef799832
Author: Mike de Boer <mdeboer@mozilla.com>
Date:   Tue Oct 10 19:14:01 2017 +0200

    Bug 1397989 - Make sure to only set non-empty values as the content-type of a document in the remote browser binding. r=mconley, a=ritu
    
    What happens is the following:
     - browser-child.js sends a statechange up to RemoteWebProgress.jsm that contains
       a `documentContentType` value of `null`, along with `requestURI` and `originalRequestURI`
       _after_ other state changes that did send a valid content-type.
     - The content-type is used by the WebProgressListener in browser.js to toggle the
       disabled state of the 'isImage' broadcaster.
     - The 'isImage' broadcaster is used by the 'cmd_find' and 'cmd_findAgain' commands to
       determine whether they should be enabled. In this case: not.
    The fix here is to _not_ set the documentContentType in the browser binding when
    it's `null`.
    
    MozReview-Commit-ID: IELoCrnOH0j
    
    --HG--
    extra : source : 4c86b1f93b8482ea3097a4edb51bb8c0e8dc67fc
---
 toolkit/modules/RemoteWebProgress.jsm | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git toolkit/modules/RemoteWebProgress.jsm toolkit/modules/RemoteWebProgress.jsm
index 04eb37d580f5..05e3dabda941 100644
--- toolkit/modules/RemoteWebProgress.jsm
+++ toolkit/modules/RemoteWebProgress.jsm
@@ -222,7 +222,11 @@ RemoteWebProgressManager.prototype = {
 
     if (isTopLevel) {
       this._browser._contentWindow = objects.contentWindow;
-      this._browser._documentContentType = json.documentContentType;
+      // Setting a content-type back to `null` is quite nonsensical for the
+      // frontend, especially since we're not expecting it.
+      if (json.documentContentType !== null) {
+        this._browser._documentContentType = json.documentContentType;
+      }
       if (typeof json.inLoadURI != "undefined") {
         this._browser.inLoadURI = json.inLoadURI;
       }
